### PR TITLE
3 Trello ticket bugs fixed

### PR DESCRIPTION
**Pull request by TheresePetersson**
* When being new primary, update sequence number to max of last_exec() or the highest sequence number of request in req_q with BOTH status PRE_PREP and PREP

* When adding client request to pend_req, check so it doesn't already exists in the list

* When being prim and assigning PRE_PREP and PREP-message, only consider unassigned client requests

### CODE REVIEW

File: modules/replication/module.py [hunk 1]
```diff
@@ -730,6 +745,12 @@ def renew_reqs(self, processors_set):
             if req in reqs_need_pre_prep:
                 # current view is equal to self.id since we are primary
                 req[REQUEST].set_view(self.id)
+                # Increment sequence number and assign
+                self.rep[self.id].inc_seq_num()
```
Comment (axelniklasson) [inline modules/replication/module.py:747]: Does this miss one seq num if there has been a reset of seq_num up around lines 630-640? Maybe we should always inc seq num after assigning it to a req instead?

File: modules/replication/module.py [hunk 2]
```diff
@@ -125,7 +124,8 @@ def run(self):
                         self.rep[self.id].get_view_changed() is False):
                     if prim_id == self.id:
                         # TODO https://bit.ly/2GAG5pc
```
Comment (axelniklasson) [inline modules/replication/module.py]: Remove this before merging to master

File: modules/replication/module.py [hunk 3]
```diff
@@ -624,6 +624,21 @@ def act_as_prim_when_view_changed(self, prim_id):
                 processor_ids.add(j)
 
         if len(processor_ids) >= (4 * self.number_of_byzantine) + 1:
+            # Update our sequence number to the most recent one.
```
Comment (axelniklasson) [inline modules/replication/module.py:625]: Should this functionality be tested in any way?

**Review by axelniklasson — APPROVED** [review]
Awesome! Some minor comments that are basically questions, let's sort them out on messenger before merging.

File: modules/replication/module.py (hunk 3 above)
Comment (TheresePetersson) [reply modules/replication/module.py:625]: Added a test for this functionality

**Review by axelniklasson — APPROVED** [review]

**Pull request merged**